src/ui/components/inputs/rules/InputRule.tsx

```diff
@@ -24,13 +22,33 @@ export const InputRule: FunctionComponent<Props> = ({
   iconSize,
   type,
   testIdSuffix,
-  noFullWidth,
+  noFullWidth = false,
 }: Props) => {
   const theme = useTheme()
-  const baseColor = type === 'Valid' ? theme.colors.greenValid : theme.colors.black
-  const color = type === 'Error' ? theme.colors.error : baseColor
+  const colorMapping: Record<InputRuleType, { text: ColorsType; icon: ColorsType }> = {
+    Valid: {
+      text: theme.designSystem.color.text.success,
+      icon: theme.designSystem.color.icon.success,
+    },
+    Error: {
+      text: theme.designSystem.color.text.error,
+      icon: theme.designSystem.color.icon.error,
+    },
+    Neutral: {
+      text: theme.designSystem.color.text.default,
+      icon: theme.designSystem.color.icon.default,
+    },
+  }
+
+  const defaultColor = {
+    text: 'text.default',
+    icon: 'icon.default',
+  }
+
+  const { text: color, icon: colorIcon } = colorMapping[type] ?? defaultColor
```

Comment:
le type `InputRuleType` est bien défini

je ne vois pas dans quel cas on pourrait tomber dnas le cas par défaut

```suggestion
  const { text: color, icon: colorIcon } = colorMapping[type]
```